powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```diff
@@ -1146,8 +1155,16 @@ public CreateActivationResponse createActivation(
                 throw localizationProvider.buildRollbackingExceptionForCode(ServiceError.INVALID_INPUT_FORMAT);
             }
 
+            // Ensure presence of the devicePublicKey
+            final String retrievedDevicePublicKey = request.getDevicePublicKey();
+            if (!StringUtils.hasText(retrievedDevicePublicKey)) {
```

Comment:
Just note that this branch is not covered by the test case (I do not know whether it is on purpose or by mistake). Not sure, but extracting a validation method could be helpful.